AgiliTEAM/pacman_env.py

```diff
@@ -0,0 +1,282 @@
+from typing import Tuple
+import cv2
+import numpy as np
+np.random.seed(0)
+
+MAP_SIZE = 30
+MAX_STEP = 100
+NUM_GHOSTS = 4
+NUM_COINS = 100
+ACTION_SPACE_SIZE = 4
+SCORE_COIN = 10
+DIRECTIONS = {"up": 0, "right": 1, "down": 2, "left": 3}
+
+
+class Pacman:
+    """
+    ------------------------
+    Pacman Environment Class
+    ------------------------
+
+
+    Attributes:
+        map_size : int
+            size of the square map's one side
+        action_space : ndarray
+            defines the executable actions in the environment
+        coins_pos : list (of tuples)
+            contains the (x, y) coordinate pairs of the generated coin objects
+        ghosts_pos : list (of tuples)
+            contains the (x, y) coordinate pairs of the generated ghost objects
+        pos : tuple
+            defines the position of PacMan
+        step_counter : int
+            keeps track of the executed steps throughout the game
+        score : int
+            keeps track of the points acquired throughout the game
+        orientation : int
+            defines PacMan's facing direction
+        state : ndarray
```

Comment:
'State' class variable is indicated in the docstring, but does not get initial value in the constructor.